src/include/optimizer/statistics/column_stats.h

```diff
@@ -1,105 +1,261 @@
 #pragma once
 
+#include <memory>
 #include <utility>
 #include <vector>
 
 #include "catalog/catalog_defs.h"
 #include "common/json_header.h"
 #include "common/macros.h"
+#include "execution/sql/value.h"
+#include "optimizer/statistics/histogram.h"
+#include "optimizer/statistics/top_k_elements.h"
 
 namespace noisepage::optimizer {
+
+/**
+ * Base class for Column Statistics
+ */
+class ColumnStatsBase {
```

Comment:
I'm still not great with templates, so if you have a better solution than using an abstract base class let me know.